src/layer/vector/Polygon.js

```diff
@@ -61,39 +61,14 @@ export var Polygon = Polyline.extend({
 		return !this._latlngs.length || !this._latlngs[0].length;
 	},
 
+	// @method getCenter(): LatLng
+	// Returns the center ([centroid](http://en.wikipedia.org/wiki/Centroid)) of the Polygon.
 	getCenter: function () {
 		// throws error when not yet added to map as this center calculation requires projected coordinates
 		if (!this._map) {
 			throw new Error('Must add layer to map before using getCenter()');
 		}
```

Comment:
Again, it will throw naturally anyway.
You can put that on the test, by the way.

Reply:
Clearer message, if it throw only `this._map is undefined` it is not clear why this error happens

Reply:
> it is not clear why this error happens

But it gets crystal clear when you see traceback.

There are A LOT of such places in leaflet, and if we age going to explicitly check every one - better switch to typescript instead.

Reply:
I don't totally agree with you. This check here makes sense because it would not be clear why `getCenter()` is not working. It is a common problem that users want to use `getCenter()` before adding to the map.

Reply:
> It is a common problem that users want to use `getCenter()` before adding to the map.

So? It fails, you see (short) call stack, you see args, and all gets clear. No?

Reply:
@mourner what do you think? Should we remove this error message from Polygon & Polyline in the `getCenter()` method?